controllers/news.controller.js

```diff
@@ -63,15 +64,32 @@ exports.getArticleById = (req, res, next) => {
       }
       res.status(200).send({ article: rows[0] });
     })
-    .catch((err) => {
-      if (err.code === "22P02") {
-        res.status(400).send({ msg: "bad request" });
-      }
-    });
+    .catch(next);
 };
 
 exports.getAllArticles = (req, res, next) => {
   getArticlesWithCommentCounts().then(({ rows }) => {
     res.status(200).send({ articles: rows });
   });
 };
+
+exports.postNewComment = (req, res, next) => {
+  if (!req.body.username || !req.body.body) {
+    return res.status(400).send({ msg: "bad request" });
+  }
```

Comment:
This really belongs in your error handling - return a Promise.reject() with an object that you can then break down inside a custom error handler